src/d3a/d3a_core/sim_results/kpi.py

```diff
@@ -76,10 +76,12 @@ def _accumulate_self_consumption_buffer(self, trade):
             self.self_consumption_buffer_wh += trade.offer.energy * 1000
 
     def _dissipate_self_consumption_buffer(self, trade):
-        if trade.seller_origin in self.ess_list and trade.seller_origin == trade.seller:
```

Comment:
I recall this check (... `and trade.seller_origin == trade.seller`) was to take into account the intermediate hierarchies. Right?

Reply:
Exactly, now this is dealt by checking the buyer_origin in each individual if clause.